qui/tray/gross_gtk3_bug_workaround.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/python3 --
+import os
+import sys
+from typing import Optional
+
+# Modifying the environment while multiple threads
+# are running leads to use-after-free in glibc, so
+# ensure that only one thread is running.
+if len(os.listdir("/proc/self/task")) != 1:
+    raise RuntimeError("threads already running")
+# If gi.override.Gdk has been imported, the GDK
+# backend has already been set and it is too late
+# to override it.
+if "gi.override.Gdk" in sys.modules:
+    raise RuntimeError("must import this module before loading GDK")
+# Only the X11 backend is supported
+os.environ["GDK_BACKEND"] = "x11"
+
+import gi
+
+gi.require_version("Gdk", "3.0")
+gi.require_version("Gtk", "3.0")
+from gi.repository import Gtk, Gdk
+
+
+is_xwayland = "WAYLAND_DISPLAY" in os.environ
+
+
+class DisgustingX11FullscreenWindowHack:
+    """
+    No-op implementation of the hack, for use on stock X11.
+    """
+
+    __slots__ = ()
+
+    @classmethod
+    def get(cls, /) -> "DisgustingX11FullscreenWindowHack":
```

Comment:
It's weird to have this as a class method here. Shouldn't it be a function outside of those classes (returning one or the other)?